database/moderation.go

```diff
@@ -571,14 +574,15 @@ func (d DB) ClaimInvite (invite string) (bool, error) {
 
   ops := []BatchQuery{
     BatchQuery{desc: "check if invite to redeem exists", stmt:"SELECT EXISTS (SELECT 1 FROM invites WHERE invite = ?)"},
+    BatchQuery{desc: "get invite code's batchid", stmt:"SELECT batchid FROM invites WHERE invite = ?"},
     BatchQuery{desc: "delete invite from table", stmt:"DELETE FROM invites WHERE invite = ?"},
   }
 
-  for _, operation := range ops {
-    operation.preparedStmt, err = tx.Prepare(operation.stmt)
-    defer operation.preparedStmt.Close()
+  for i, operation := range ops {
+    ops[i].preparedStmt, err = tx.Prepare(operation.stmt)
+    defer ops[i].preparedStmt.Close()
```

Comment:
classic golang. i think i came up with a good mnemonic to reduce errors in this department.

if ranging over e.g. a slice and wanting to change something, never use the **second** parameter of range.

```golang
for i, _ := range items { items[i].member = ... }
```

while, if ranging over e.g. a slice and only wanting to consume items, never use the **first** parameter of range:



```golang
for _, item := range items { use(item) }
```